sdks/uniswapx-sdk/src/order/PriorityOrder.ts

```diff
@@ -0,0 +1,376 @@
+      outputs: scaleOutputs(this.info.outputs, options.priorityFee),
+    };
+  }
+
+  /**
+   * Returns the parsed validation
+   * @return The parsed validation data for the order
+   */
+  get validation(): CustomOrderValidation {
+    return parseValidation(this.info);
+  }
+
+  private toPermit(): PermitTransferFrom {
+    return {
+      permitted: {
+        token: this.info.input.token,
+        amount: this.info.input.amount,
+      },
+      spender: this.info.reactor,
+      nonce: this.info.nonce,
+      deadline: this.info.deadline,
+    };
+  }
+
+  private witnessInfo(): PriorityWitnessInfo {
+    return {
+      info: {
+        reactor: this.info.reactor,
+        swapper: this.info.swapper,
+        nonce: this.info.nonce,
+        deadline: this.info.deadline,
+        additionalValidationContract: this.info.additionalValidationContract,
+        additionalValidationData: this.info.additionalValidationData,
+      },
+      startBlock: this.info.startBlock,
+      input: this.info.input,
+      outputs: this.info.outputs,
+    };
+  }
+
+  private witness(): Witness {
+    return {
+      witness: this.witnessInfo(),
+      witnessTypeName: "PriorityOrder",
+      witnessType: PRIORITY_ORDER_TYPES,
+    };
+  }
+}
+
+function parseSerializedOrder(serialized: string): PriorityOrderInfo {
+  const abiCoder = new ethers.utils.AbiCoder();
+  const decoded = abiCoder.decode(PRIORITY_ORDER_ABI, serialized);
+  const [
+    [
+      [
+        reactor,
+        swapper,
+        nonce,
+        deadline,
+        additionalValidationContract,
+        additionalValidationData,
+      ],
+      startBlock,
+      [token, amount, mpsPerPriorityFeeWei],
+      outputs,
+    ],
+  ] = decoded;
+
+  return {
+    reactor,
+    swapper,
+    nonce,
+    deadline: deadline.toNumber(),
+    additionalValidationContract,
+    additionalValidationData,
+    startBlock,
+    input: {
+      token,
+      amount,
+      mpsPerPriorityFeeWei,
+    },
+    outputs: outputs.map(
+      ([token, amount, mpsPerPriorityFeeWei, recipient]: [
+        string,
+        BigNumber,
+        BigNumber,
+        string
+      ]) => {
+        return {
+          token,
+          amount,
+          mpsPerPriorityFeeWei,
+          recipient,
+        };
+      }
+    ),
+  };
+}
+
+function scaleInput(input: PriorityInput, priorityFee: BigNumber): BigNumber {
+  if (priorityFee.mul(input.mpsPerPriorityFeeWei).gte(MPS)) {
+    return BigNumber.from(0);
+  }
+  return input.amount
+    .mul(MPS.sub(priorityFee.mul(input.mpsPerPriorityFeeWei)))
+    .div(MPS);
+}
+
+function scaleOutputs(
+  outputs: PriorityOutput[],
+  priorityFee: BigNumber
+): PriorityOutput[] {
+  return outputs.map((output) => {
+    const product = output.amount.mul(
+      MPS.add(priorityFee.mul(output.mpsPerPriorityFeeWei))
+    );
+    const mod = product.mod(MPS);
+    const div = product.div(MPS);
+    return {
+      ...output,
+      amount: mod.eq(0) ? div : div.add(1),
```

Comment:
do you need to do this rounding for the input as well? for round down

Reply:
I think BigNumber.div() rounds down by default